source/game/kart/KartPhysics.cc

```diff
@@ -66,9 +93,21 @@ void KartPhysics::setVelocity(const EGG::Vector3f &vel) {
     m_velocity = vel;
 }
 
+void KartPhysics::set_fc(f32 val) {
+    m_fc = val;
+}
+
 KartPhysics *KartPhysics::Create(const KartParam &param) {
-    // TODO: Hitboxes and BSP params
-    return new KartPhysics(param.isBike());
+    KartPhysics *physics = new KartPhysics(param.isBike());
+
+    const BSP &bsp = param.bsp();
+
+    physics->set_fc(physics->hitboxGroup()->initHitboxes(bsp.hitboxes));
```

Comment:
`m_fc` is initialized to `50.0f` in the constructor in-game... But it is immediately overwritten in this `CreateInstance` function. When we run into circumstances like this, is it okay to skip the assignment in the constructor?

Reply:
> `m_fc` is initialized to `50.0f` in the constructor in-game... But it is immediately overwritten in this `CreateInstance` function. When we run into circumstances like this, is it okay to skip the assignment in the constructor? Currently I don't set it in the c'tor in this PR

I want to give my two cents on this: I wouldn't do optimizations like this, even if they are obvious. When extending code in HanachanC for further parity, I found that 'obvious' optimizations that were applied like this caused trouble when more of the code was reversed and there turned out to be some edge case where the assignment order mattered. I assume in this case it wouldn't matter though as the whole routine is already decompiled (?).

Another consideration is optimizations like this can also be annoying to match up when watching values in the actual game. There would be a desynced value in mid-simulation versus the game, but it would synchronize at a later point. I've had a few time where this scenario happened and I wasted my time looking at why the value was wrong, when it turned out to be an optimization to remove a redundant assignment and it was correct in the end.

So for the sake of consistency, maybe best to keep these oversights, but perhaps leave a comment that this is the case, like decompile projects in other games tend to do.

Reply:
> I want to give my two cents on this: I wouldn't do optimizations like this, even if they are obvious.

My thoughts exactly.

Reply:
@ficool2 Great points, thank you for offering that perspective. OK I believe we are in agreement, even if an assignment appears "pointless", we should still include it, but leave a comment. I will audit this entire PR to verify variables such as this example are not skipped in constructor bodies.